pandas/io/common.py

```diff
@@ -519,6 +519,21 @@ def infer_compression(
     raise ValueError(msg)
 
 
+def check_parent_directory(fpath: str):
+    """
+    Check if parent directory of a file exists, raise OSError if it does not
+
+    Parameters
+    ----------
+    fpath: str
+        File path
+
+    """
+    dirname = os.path.dirname(fpath)
+    if len(dirname) != 0 and not os.path.isdir(dirname):
```

Comment:
Might look slightly simpler with pathlib: `if not Path(fpath).parent.is_dir():`

Reply:
done